spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/ModuleSerializable.scala

```diff
@@ -285,6 +289,43 @@ trait ModuleSerializable extends Loadable with Savable{
    */
   protected def copy2BigDL[T: ClassTag](context: DeserializeContext, module : ModuleData[T])
                                        (implicit ev: TensorNumeric[T]): Unit = {
+
+    if (context.bigdlModule.getHasParameters) {
+      copyParameters2BigDL(context, module)
+    } else {
+      // for legacy format models
+      copyWeightAndBias(context, module)
+    }
+  }
+
+  private def copyParameters2BigDL[T: ClassTag]
+    (context: DeserializeContext, module : ModuleData[T])
+    (implicit ev: TensorNumeric[T]): Unit = {
+
+    val serializedParameters = context.bigdlModule.getParametersList.asScala.toArray
+
+    val arrayValue = ArrayValue.newBuilder
+    arrayValue.setDatatype(DataType.TENSOR)
+    serializedParameters.foreach(param => arrayValue.addTensor(param))
+    arrayValue.setSize(serializedParameters.length)
```

Comment:
What will happen if we set `hasParameters` to true, but without the parameters?

Reply:
this variable will only be set to true when model has parameters(false by default), check below code : 
```scala
if (parameters != null && parameters._1 != null) {
      modelBuilder.setHasParameters(true)
      parameters._1.foreach(parameter => {
        val tensorAttr = AttrValue.newBuilder
        TensorConverter.setAttributeValue(context, tensorAttr, parameter)
        modelBuilder.addParameters(tensorAttr.getTensorValue)
      })
}
```